packages/launchpad/src/use-tracking/index.tsx

```diff
@@ -0,0 +1,67 @@
+import { recordTracksEvent } from '@automattic/calypso-analytics';
+import { useEffect, useMemo } from 'react';
+import type { Task } from '../types';
+import type { SiteDetailsOptions } from '@automattic/data-stores';
+
+interface LogParams {
+	tasks: Task[];
+	checklistSlug: string | null;
+	context: string;
+	siteIntent: SiteDetailsOptions[ 'site_intent' ];
+}
+
+export const useTracking = ( params: LogParams ) => {
+	const { tasks, checklistSlug, context, siteIntent } = params;
+	const completedSteps = useMemo( () => tasks.filter( ( task ) => task.completed ), [ tasks ] );
+	const taskNames = useMemo( () => tasks.map( ( task ) => task.id ).join( ',' ), [ tasks ] );
```

Comment:
There is a small bug here: the original code included a `,` prefix and suffix so you could search for a specific value via `LIKE '%,my_task_id,%'` regardless of which position the task was in.
```suggestion
	const taskNames = useMemo( () => `,${ tasks.map( ( task ) => task.id ).join( ',' ) },`, [ tasks ] );
```

Reply:
I updated the code to make it more explicit and added a test 598b08238290d8073d118389d4ee3acfc83b2d05 to protect us from future changes